### PR TITLE
spack.bat: Fixup CL arg parsing

### PR DESCRIPTION
**Pull request by johnwparent**
Previous changes to this file stopped directly processing CL args to stop batch `for` from interpolating batch reserved characters needed in arguments like URLS. In doing so, we relied on `for` for an easy "split" operation, however this incorrectly splits paths with spaces in certain cases that would otherwise be handled without issue allowing the `for` loop more control or handling CL args directly through numbered variables (`%1`). Processing everything ourselves with manual looping via `goto` statements allows for full control over CL parsing and handling of both paths with spaces and reserved characters.

### CODE REVIEW

File: bin/spack.bat [hunk 1]
```diff
@@ -51,65 +51,43 @@ setlocal enabledelayedexpansion
...
+) else (
+    if not defined _sp_args (
+        set "_sp_args=!t!"
+    ) else (
+        set "_sp_args=!_sp_args! !t!"
+    )
 )
-rem  if this is not nil, we have more tokens to process
+
+rem  if this is not nu;ll, we have more tokens to process
```
Comment (scheibelp) [inline bin/spack.bat:90]: Typo ("nu;ll")

File: bin/spack.bat [hunk 2]
```diff
@@ -51,65 +51,43 @@ setlocal enabledelayedexpansion
...
-        shift
-    ) else (
+rem Set first cl argument (denoted by %1) to be processed
+set t=%1
+rem shift moves all cl positional arguments left by one
+rem meaning %2 is now %1, this allows us to iterate over each
+rem argument
+shift
+rem assign next "first" cl argument to cl_args, will be null when
+rem there are now further arguments to process
```
Comment (scheibelp) [inline bin/spack.bat:63]: (typo) "now" -> "no"

**Review by scheibelp — APPROVED** [review]
The changes look good (I only see minor typos) but it would be nice to create a testing script similar to `unit-tests.yaml -> run-shell-tests -> setup-env-test.sh` where we can pass arguments to `spack.bat` and make sure it parses them into command invocations properly.

I don't see anything like this the direct analog of `spack.bat` - `bin/spack` - so I'm not inclined to make that request a blocker. Let's say though that if we find ourselves back here soon (other than to address comments/typos), we talk about making a test to avoid regressions?

**Comment by johnwparent** [comment]
> The changes look good (I only see minor typos) but it would be nice to create a testing script similar to `unit-tests.yaml -> run-shell-tests -> setup-env-test.sh` where we can pass arguments to `spack.bat` and make sure it parses them into command invocations properly.
> 
> I don't see anything like this the direct analog of `spack.bat` - `bin/spack` - so I'm not inclined to make that request a blocker. Let's say though that if we find ourselves back here soon (other than to address comments/typos), we talk about making a test to avoid regressions?

I actually have an open PR that refactors this entry point again (an attempt to make things the same as unix/powershell on CMD) so perhaps that PR is a good place to discuss a test? I agree that having one is fairly important

**Pull request merged**